src/main/java/edu/oregonstate/cs361/battleship/BattleshipModel.java

```diff
@@ -269,85 +274,90 @@ public boolean shipInBounds(Coordinate c, int orientation, int length){
-            Coordinate end = new Coordinate(endAcross, endDown);
-            testShip = new MilitaryShip("test", size, start, end, false);
-            for(int i = 0; i < playerShipPoints.size(); i++){
-                if(testShip.covers(playerShipPoints.get(i)))
-                    return "Placement overlaps another ship";
-            }
-            Coordinate[] myPoints = new Coordinate[size];
-            for(int i = 0; i < size; i++){
-                Coordinate toAdd = new Coordinate( Across, Down + i);
-                playerShipPoints.add(toAdd);
-                myPoints[i] = toAdd;
-            }
-            if(shipName.equalsIgnoreCase("aircraftCarrier") || shipName.equalsIgnoreCase("battleship") || shipName.equalsIgnoreCase("submarine")){
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip) getShip(shipName);
-                militaryPlaceIndex++;
-            }
-            else{
-                getShip(shipName).setLocation(start, end);
-                getShip(shipName).setPoints(myPoints);
-                playerCivilianShips[civilianPlaceIndex] = (CivilianShip) getShip(shipName);
-                civilianPlaceIndex++;
-            }
         }
 
+        //Return error string if ship would go off board
+        if(endAcross > BOARD_SIZE || endDown > BOARD_SIZE)
+            return "Ship Placement out of bounds";
+
+        //Create new start and end coordinates for ship
+        start = new Coordinate(Across, Down);
+        end = new Coordinate(endAcross, endDown);
+        testShip = new MilitaryShip("test", size, start, end, false);
+
+        //Test if ship position would overlap any other ship
+        for(int i = 0; i < playerShipPoints.size(); i++){
+            if(testShip.covers(playerShipPoints.get(i)))
+                return "Placement overlaps another ship";
+        }
+
+        //Add ship coordinates to playerShipPoints
+        for(int i = 0; i < size; i++){
+            //If vertical, move along across or row
+            if (orientation.equals("vertical"))
+                toAdd = new Coordinate(Across + i, Down);
+            //If horizontal, move along Down or column
+            else if (orientation.equals("horizontal"))
+                toAdd = new Coordinate(Across,Down+i);
+
+            playerShipPoints.add(toAdd);
+            myPoints[i] = toAdd;
+        }
+
+        //Set location of ship, assign points, and then add to corresponding player ship array
+        placingShip.setLocation(start,end);
+        placingShip.setPoints(myPoints);
+
+        //Military
+        if (placingShip instanceof MilitaryShip) {
+            playerMilitaryShips[militaryPlaceIndex] = (MilitaryShip)placingShip;
+            militaryPlaceIndex++;
+        }
+        //Civilian
+        else if (placingShip instanceof CivilianShip) {
```

Comment:
This probably could just be an else statement. Its fine either way, but that might be a bit cleaner if you change it.